src/sisl/io/vasp/out.py

```diff
@@ -144,6 +145,73 @@ def readE(itt):
             return E[-1]
         return None
 
+
+    @sile_fh_open()
+    def read_trajectory(self, all=True):
+        """ Reads the cell+atomic positions and forces from OUTCAR (in Ang and eV/Ang, respectively)
+
+        Parameters
+        ----------
+        all: bool, optional
+            whether to return the whole trajectory or just the last step
+
+        Returns
+        -------
+        PropertyDict or list : cell, atomic positions, and forces (optionally from each step)
+        """
+
+        def read_step(itt):
+            # read cell vectors
+            f = self.step_to("BASIS-vectors are now", allow_reread=False)[0]
+            if not f:
+                return None
+            for i in range(4):
+                next(itt) # skip four lines
+            C = []
+            for i in range(3):
+                line = next(itt)
+                v = line.split()
+                # direct lattice vectors
+                C.append([float(v[0]), float(v[1]), float(v[2])])
+            # read a position-force table
+            f = self.step_to("TOTAL-FORCE (eV/Angst)", allow_reread=False)[0]
+            if not f:
+                return None
+            next(itt) # -----
+            P, F = [], []
+            line = next(itt)
+            while "----" not in line:
+                v = line.split()
+                # positions and forces
+                P.append([float(v[0]), float(v[1]), float(v[2])])
+                F.append([float(v[3]), float(v[4]), float(v[5])])
```

Comment:
It'll probably be faster to do the final conversion in `np.array(C,dtype=np.float64)` or similar, then the code becomes cleaner and faster.

Reply:
Done

Reply:
Thanks!